docs/resources/server/advancements.md

```diff
@@ -92,48 +89,42 @@ public boolean matches(ItemStack stack) {
 
 ### SimpleCriterionTrigger
 
-The `SimpleCriterionTrigger<T>` subclass, where `T` is the type of the trigger instance, is responsible for specifying the registry name of the trigger, creating a trigger instance, and a method to check trigger instances and run attached listeners on success.
-
-The registry name of the trigger is supplied to `#getId`. This should match the registry name supplied to the trigger instance.
+The `SimpleCriterionTrigger<T>` subclass, where `T` is the type of the trigger instance, is responsible for specifying a codec to serialize `T`s, and supplying a method to check trigger instances and run attached listeners on success.
 
-A trigger instance is created via `#createInstance`. This method reads a criteria from JSON.
+The codec is typically defined as a constant of the inner `SimpleCriterionTrigger.SimpleInstance` class, and simply returned by the trigger's `codec()` method.
```

Comment:
Wasn't this mentioned above as well? Where the codec lives should probably be restructured in this doc.